processor/transformprocessor/README.md

```diff
@@ -8,66 +8,146 @@
 | Warnings                 | [Unsound Transformations, Identity Conflict, Orphaned Telemetry, Other](#warnings) |
 
 The transform processor modifies telemetry based on configuration using the [OpenTelemetry Transformation Language](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl).
-The processor takes a list of statements for each signal type and executes the statements against the incoming telemetry in the order specified in the config.  Each statement can access and transform telemetry using functions and allow the use of a condition to help decide whether the function should be executed.
+
+For each signal type, the processor takes a list of statements associated to a [Context type](#contexts) and executes the statements against the incoming telemetry in the order specified in the config.
+Each statement can access and transform telemetry using functions and allow the use of a condition to help decide whether the function should be executed.
 
 ## Config
 
-The transform processor allows configuring statements for traces, metrics, and logs. Each signal specifies a list of string statements that get passed to the OTTL for interpretation.
+The transform processor allows configuring multiple context statements for traces, metrics, and logs.
+The value of `context` specifies which [OTTL Context](#contexts) to use when interpreting the associated statements.
+The statement strings, which must be OTTL compatible, will be passed to the OTTL and interpreted using the associated context. 
+Each context will be processed in the order specified and each statement for a context will be executed in the order specified.
 
 ```yaml
 transform:
-  <traces|metrics|logs>:
-    statements:
-      - string
-      - string
-      - string
+  <trace|metric|log>_statements:
+    - context: string
+      statements:
+        - string
+        - string
+        - string
+    - context: string
+      statements:
+        - string
+        - string
+        - string
 ```
 
+Proper use of contexts will provide increased performance and capabilities.  See [Contexts](#contexts) for more details.
+
+Valid values for `context` are:
+
+| Signal            | Context Values                                 |
+|-------------------|------------------------------------------------|
+| trace_statements  | `resrouce`, `scope`, `trace`, and `spanevent`  |
```

Comment:
Not necessarily to be resolved here, but should this be `span` instead of `trace`?

Reply:
yes, I plan to rename trace context to span context soon.